### DISCUSSION TITLE
Getting error while building the project

### DISCUSSION THREAD

**Discussion by Asif047** [Kaaveh/ComposeNews#70] [General]
I just cloned your project for checking that with this updated architecture. But while building the project, I am getting the following error:

A problem occurred configuring project ':app'.
> Could not determine the dependencies of null.
   > Could not resolve all task dependencies for configuration ':app:classpath'.
      > Could not resolve project :build-logic:convention.
        Required by:
            project :app
         > No matching variant of project :build-logic:convention was found. The consumer was configured to find a library for use during runtime, compatible with Java 11, packaged as a jar, and its dependencies declared externally, as well as attribute 'org.gradle.plugin.api-version' with value '8.1' but:
             - Variant 'apiElements' capability ir.composenews.composenews.buildlogic:convention:unspecified declares a library, packaged as a jar, and its dependencies declared externally:
                 - Incompatible because this component declares a component for use during compile-time, compatible with Java 17 and the consumer needed a component for use during runtime, compatible with Java 11
                 - Other compatible attribute:
                     - Doesn't say anything about org.gradle.plugin.api-version (required '8.1')
             - Variant 'mainSourceElements' capability ir.composenews.composenews.buildlogic:convention:unspecified declares a component, and its dependencies declared externally:
                 - Incompatible because this component declares a component of category 'verification' and the consumer needed a library
                 - Other compatible attributes:
                     - Doesn't say anything about its target Java version (required compatibility with Java 11)
                     - Doesn't say anything about its elements (required them packaged as a jar)
                     - Doesn't say anything about org.gradle.plugin.api-version (required '8.1')
                     - Doesn't say anything about its usage (required runtime)
             - Variant 'runtimeElements' capability ir.composenews.composenews.buildlogic:convention:unspecified declares a library for use during runtime, packaged as a jar, and its dependencies declared externally:
                 - Incompatible because this component declares a component, compatible with Java 17 and the consumer needed a component, compatible with Java 11
                 - Other compatible attribute:
                     - Doesn't say anything about org.gradle.plugin.api-version (required '8.1')
             - Variant 'testResultsElementsForTest' capability ir.composenews.composenews.buildlogic:convention:unspecified:
                 - Incompatible because this component declares a component of category 'verification' and the consumer needed a library
                 - Other compatible attributes:
                     - Doesn't say anything about how its dependencies are found (required its dependencies declared externally)
                     - Doesn't say anything about its target Java version (required compatibility with Java 11)
                     - Doesn't say anything about its elements (required them packaged as a jar)
                     - Doesn't say anything about org.gradle.plugin.api-version (required '8.1')
                     - Doesn't say anything about its usage (required runtime)
                     
                     ------------------------------------------------------
                     
                     
                     Can you please help me out to solve this?

**Comment by Kaaveh** [comment]
I think you must set Gradle JDK to 17 in your Android studio. Let me know if it's not solved.

![Screenshot 2023-08-28 at 4 06 36 PM](https://github.com/Kaaveh/ComposeNews/assets/39463024/2fbd43a5-9ef7-4ade-822b-567f9fc846c3)
